src/libpmemfile-posix/dir.c

```diff
@@ -229,10 +229,9 @@ vinode_add_dirent(PMEMfilepool *pfp,
 	struct pmemfile_inode *parent = parent_vinode->inode;
 
 	/* don't create files in deleted directories */
-	if (parent->nlink == 0)
```

Comment:
I don't think this is a problem, I use similar constructs sometimes.
I would just add curly brackets for the outer ```if```